### PR TITLE
ArangoDb Persister and Store are created

### PR DESCRIPTION
**Pull request by Perfect111**
ArangoDb Persister and Store are created

### CODE REVIEW

File: py2store/persisters/_arangodb_in_progress.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    >>> s[{'name': 'vitalik'}] = {'yob': 1994, 'proj': 'ethereum', 'bdfl': True}
+    >>> s[{'name': 'vitalik'}]
+    {'yob': 1994, 'proj': 'ethereum', 'bdfl': True}
+    >>> for key, val in s.items():
+    ...     print(f"{key}: {val}")
+    {'name': 'guido'}: {'yob': 1956, 'proj': 'python', 'bdfl': False}
+    {'name': 'vitalik'}: {'yob': 1994, 'proj': 'ethereum', 'bdfl': True}
+    """
+
+    def clear(self):
```
Comment (Kulv3r) [inline py2store/persisters/_arangodb_in_progress.py]: This method is defined in the parent class, are you sure you need to redefine it here?

File: py2store/persisters/_arangodb_in_progress.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+        self._key_fields = key_fields
+
+    def __getitem__(self, k):
+        reserved = {"_key", "_id", "_rev"}
+        f = self._collection.fetchFirstExample(k)
+        if f is not None and len(f) == 1:
+            d = f[0].getStore()
+            # exclude reserved keys and corresponded values
+            d = {x: d[x] for x in d if x not in reserved and x not in self._key_fields}
```
Comment (Kulv3r) [inline py2store/persisters/_arangodb_in_progress.py]: Makes sense to separate fetching and excluding so you can re-use fetching later (like in __delitem__).

File: py2store/persisters/_arangodb_in_progress.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+    def __delitem__(self, k):
+        if len(k) > 0:
+            f = self._collection.fetchFirstExample(k)
+            if f is not None:
+                return f[0].delete()
+            else:
+                raise KeyError(f"You can't removed that key: {k}")
+        else:
+            raise KeyError(f"You can't removed that key: {k}")
```
Comment (Kulv3r) [inline py2store/persisters/_arangodb_in_progress.py]: Is there a way to rewrite it without duplicating exception raising code?

File: py2store/persisters/_arangodb_in_progress.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+    def __iter__(self):
+        docs = self._collection.fetchAll()
+        k = []
+        for d in docs:
+            k = k + [{x: d[x] for x in d.getStore() if x in self._key_fields}]
+
+        yield from k
```
Comment (Kulv3r) [inline py2store/persisters/_arangodb_in_progress.py]: Since `k` is a list, what is a purpose of `yield`ing it?

Comment (Perfect111) [reply py2store/persisters/_arangodb_in_progress.py]: yield from k returns a generator. If use 
`    def __iter__(self):
        docs = self._collection.fetchAll()
        k = []
        for d in docs:
            k = k + [{x: d[x] for x in d.getStore() if x in self._key_fields}]

        return k`
calling
`for _key in s:
    del s[_key]
`
we get
`Traceback (most recent call last):
  File "<input>", line 1, in <module>
TypeError: iter() returned non-iterator of type 'list'`

Comment (Perfect111) [reply py2store/persisters/_arangodb_in_progress.py]: anyway, I improved this method in new PR

**Pull request merged**